sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroSource.java

```diff
@@ -482,51 +486,56 @@ public AvroReader(AvroSource<T> source) {
       return (AvroSource<T>) super.getCurrentSource();
     }
 
+    // Precondition: the stream is positioned after the sync marker in the current (about to be
+    // previous) block. currentBlockSize equals the size of the current (previous) block, or zero
+    // if this reader was just started.
+    //
+    // Postcondition: same as above, but for the current (formerly next) block.
```

Comment:
This is helpful, but some of the parantheticals seem confusing.

"current (previous)" I think should be "current (about to be previous)" or just say "current" since you've already stated that the current block in the precondition is about to be previous.

Similarly, in Postcondition: "current (formerly next)"  could be clarified to the "new current (formerly next)"